terminal-api/utils.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  CustomerTerminalStateTypes, TransactionCancelStateTypes, CustomerUidAndDiscount
+} from '../types/transaction.ts'
+import { ConfigurationSchema } from '../types/config.ts'
+import { ILogger } from "./logger.ts";
+
+// const BEARER_TOKEN: string = coffee.get("settings.bearer_token").string();
+// const POS_ID: string = coffee.get("settings.pos_id").string();
+// const URL: string = coffee.get("settings.base_url").string() +
+//   coffee.get("settings.terminal_id").string() + "/";
+
+
+export function generateIds(): [string, string] {
+  return [globalThis.crypto.randomUUID(), globalThis.crypto.randomUUID()];
+}
+
+export async function httpRequest(
+  endpoint: string,
+  method: string,
+  body: string | null,
+  config: ConfigurationSchema,
+  timeout = 120000
+): Promise<Response> {
+  let response: Response;
+
```

Comment:
The "delay" should be put in here and removed from the other spots. We want the delay to slow all calls using httpRequest()